File: valo_tui/screens/splash.py
"""Landing page — ASCII logo, version, cache freshness. Press enter to enter
the app (or it auto-advances after a few seconds)."""

from __future__ import annotations

import pyfiglet

from textual.app import ComposeResult
from textual.binding import Binding
from textual.containers import Center, Middle
from textual.screen import Screen
from textual.widgets import Static

from .. import __version__
from ..data import cache

ART = pyfiglet.figlet_format("valo-tui", font="slant")


class SplashScreen(Screen):
    BINDINGS = [
        Binding("enter,space", "enter_app", "Enter", show=False),
        Binding("q", "app.quit", "Quit", show=False),
    ]

    CSS = """
    SplashScreen { background: #0a1822; align: center middle; }
    #logo { color: #e8674e; text-style: bold; width: auto; }
    #tag  { color: #4a708b; width: auto; margin-top: 1; }
    #freshness { color: #4a708b; width: auto; margin-top: 1; }
    #enter-hint { color: #c8d8e8; width: auto; margin-top: 2; }
    """

    def compose(self) -> ComposeResult:
        with Middle():
            with Center():
                yield Static(ART, id="logo")
            with Center():
                yield Static("valorant esports in your terminal", id="tag")
            with Center():
                yield Static("", id="freshness")
            with Center():
                yield Static("[ press enter ]", id="enter-hint")

    def on_mount(self) -> None:
        ts = cache.last_updated()
        fresh = f"cache · {ts} UTC" if ts else "cache empty — start the worker"
        self.query_one("#freshness", Static).update(f"v{__version__}  ·  {fresh}")
        # auto-advance as a fallback so it never gets stuck
        self.set_timer(6.0, self.action_enter_app)

    def action_enter_app(self) -> None:
        if self.is_running:
            self.dismiss()

File: internal/screens/splash.go
package screens

import (
	"time"

	tea "charm.land/bubbletea/v2"
	"charm.land/lipgloss/v2"

	"github.com/jashkarangiya/valo-tui/internal/data"
	"github.com/jashkarangiya/valo-tui/internal/styles"
)

// version is shown on the splash and about page. Kept here until a build-time
// ldflags value replaces it.
const version = "0.2.0"

// "valo-tui" in figlet ansi_shadow (the same font family as the match-detail
// score hero), so the branding is consistent across the app.
const logoArt = `██╗   ██╗ █████╗ ██╗      ██████╗    ████████╗██╗   ██╗██╗
██║   ██║██╔══██╗██║     ██╔═══██╗   ╚══██╔══╝██║   ██║██║
██║   ██║███████║██║     ██║   ██║█████╗██║   ██║   ██║██║
╚██╗ ██╔╝██╔══██║██║     ██║   ██║╚════╝██║   ██║   ██║██║
 ╚████╔╝ ██║  ██║███████╗╚██████╔╝      ██║   ╚██████╔╝██║
  ╚═══╝  ╚═╝  ╚═╝╚══════╝ ╚═════╝       ╚═╝    ╚═════╝ ╚═╝`

// Splash is the landing page: logo, tag, cache freshness, enter hint.
type Splash struct {
	w, h  int
	fresh string
}

// NewSplash builds the splash, reading cache freshness up front.
func NewSplash(w, h int) Splash {
	ts := data.LastUpdated()
	fresh := "cache empty — start the worker"
	if ts != "" {
		fresh = "cache · " + ts + " UTC"
	}
	return Splash{w: w, h: h, fresh: "v" + version + "  ·  " + fresh}
}

func (s *Splash) SetSize(w, h int) { s.w, s.h = w, h }

func (s Splash) Init() tea.Cmd {
	// Auto-advance as a fallback so it never gets stuck.
	return tea.Tick(6*time.Second, func(time.Time) tea.Msg { return EnterAppMsg{} })
}

func (s Splash) Update(msg tea.Msg) (Splash, tea.Cmd) {
	switch msg := msg.(type) {
	case tea.KeyPressMsg:
		switch msg.String() {
		case "enter", " ", "space":
			return s, func() tea.Msg { return EnterAppMsg{} }
		}
	}
	return s, nil
}

func (s Splash) View() string {
	logo := lipgloss.NewStyle().Foreground(styles.Accent).Bold(true).Render(logoArt)
	tag := lipgloss.NewStyle().Foreground(styles.Muted).Render("valorant esports in your terminal")
	fresh := lipgloss.NewStyle().Foreground(styles.Muted).Render(s.fresh)
	hint := lipgloss.NewStyle().Foreground(styles.Text).Render("[ press enter ]")

	body := lipgloss.JoinVertical(lipgloss.Center, logo, "", tag, fresh, "", hint)
	return lipgloss.Place(s.w, s.h, lipgloss.Center, lipgloss.Center, body)
}
